### DISCUSSION TITLE
Add macros to support C++ Properties (like on C#)

### DISCUSSION THREAD

**Discussion by aismann** [axmolengine/axmol#850] [Q&A] [answered]
See https://www.codeproject.com/articles/118921/c-properties

@halx99 what are you mean?

**Comment by halx99** [comment]
No idea.

**Comment by rh101** [comment, marked as answer]
Adding something like this would cause a lot more problems than it solves.  It would also make debugging harder, as macros always do.

It's been done before in Cocos2d-x, and eventually their usage was removed.  Just look at `CCPlatformMacros.h`, with an example being `CC_PROPERTY`:

```
#define CC_PROPERTY(varType, varName, funName) \
protected:                                     \
    varType varName;                           \
                                               \
public:                                        \
    virtual varType get##funName() const;      \
    virtual void set##funName(varType var);
```

If you search for `CC_PROPERTY` in the platform, you won't find any usage of it. 

Please don't complicate the engine code by using macros like that.

**Reply by aismann** [reply]
> Please don't complicate the engine code by using macros like that.

A good point. Thanks.
